chain/network/src/view_client_adapter.rs

```diff
@@ -0,0 +1,79 @@
+use futures::future::BoxFuture;
+use futures::Future;
+use near_network_primitives::types::{NetworkViewClientMessages, NetworkViewClientResponses};
+use std::fmt;
+use std::sync::Arc;
+
+/// [`ViewClientAdapter`] defines the subset of view client's interface which is
+/// needed by the network.
+///
+/// It exists primarily to break inter-crate dependencies -- network should not
+/// know the specific type of the client.
```

Comment:
I also think that "Handler" is more appropriate than Adapter, or actually it should be even "Server" as ViewClient + Client in this context serve as an implementation for the RPC Server API defined by the near-network crate. In this terminology, the PeerManagerAdapter (which I assume you wanted to stay consistent with) serves as an RPC Client, so a (Network)Client would be a consistent name.

Reply:
Yeah, that's debatable! I guess I'd leave this as Adapter for this PR, and leave it up to you where you want rename both this and peer manager. 

Also, I must confess I don't fully understand the Server/Client suggestion. What would be the name? just `ViewClient`? or `ViewClientClient`? That's why I like `Handler` -- given that this is in the same process, it's imporant to distinguish the actual thing which runs on some thread, from the interface one uses to inrerract with it. In synchronous programming, both are the same X. In async, I usually use `X` (or `XActor`) for the thing, and `XHandle` for the public API